pkg/controller/common/helpers.go

```diff
@@ -361,39 +361,34 @@ func ValidateMachineConfig(cfg mcfgv1.MachineConfigSpec) error {
 // IgnParseWrapper parses rawIgn for both V2 and V3 ignition configs and returns
 // a V2 or V3 Config or an error. This wrapper is necessary since V2 and V3 use different parsers.
 func IgnParseWrapper(rawIgn []byte) (interface{}, error) {
-	ignCfgV3_2, rptV3_2, errV3_2 := ign3.Parse(rawIgn)
-	if errV3_2 == nil && !rptV3_2.IsFatal() {
+	// ParseCompatibleVersion will parse any config <= N to version N
+	ignCfgV3, rptV3, errV3 := ign3_3.ParseCompatibleVersion(rawIgn)
+	if errV3 == nil && !rptV3.IsFatal() {
+		// Regardless of the input version it has now been translated to a 3.3 config, downtranslate to 3.2 to match
+		// what is used internally
+		ignCfgV3_2, errV3_2 := v33tov32.Translate(ignCfgV3)
+		if errV3_2 != nil {
+			// This case should only be hit if fields that only exist in v3_3 are being used which are not
+			// currently supported by the MCO
+			return ign3types.Config{}, errors.Errorf("translating Ignition config failed with error %v", errV3_2)
```

Comment:
wondering if it's worth being explicit in the error message like in the comment?